tests/unit/data_test.py

```diff
@@ -28,42 +27,40 @@ def test_full_drop_na(fake_testing_dataset):
     """
     must drop 11 (9 in my_feature, 2 other in time_til_outcome)
     """
-    fake_testing_dataset._remove_na(ignore=[])
+    fake_testing_dataset.remove_na(ignore=[])
     assert len(fake_testing_dataset) == 9
 
 
 def test_part_drop_na(fake_testing_dataset):
     """
     must drop 9 (9 in my_feature, nothing from time_til_outcome, as the field is in ignore)
     """
-    fake_testing_dataset._remove_na(ignore=["time_til_outcome"])
+    fake_testing_dataset.remove_na(ignore=["time_til_outcome"])
     assert len(fake_testing_dataset) == 11
 
 
 def test_fill_defaults_and_drop_na(fake_testing_dataset):
     """
     must replace 9 (None->0 for my_feature) and drop 2 (from time_til_outcome)
     """
-    fake_testing_dataset._replace_missing_data(defaults_map={"my_feature": 0})
-    fake_testing_dataset._remove_na(ignore=[])
+    fake_testing_dataset.replace_missing_data(defaults_map={"my_feature": 0})
+    fake_testing_dataset.remove_na(ignore=[])
     assert len(fake_testing_dataset) == 18
 
 
 def test_remove_strong_signals(fake_testing_dataset):
     """
     must drop 5 (4 0s and 1 (-2)s)
     """
-    fake_testing_dataset._remove_strong_signals()
+    fake_testing_dataset.remove_strong_signals()
     assert len(fake_testing_dataset) == 15
 
 
-def test_prepare_data(fake_testing_dataset):
+def test_chained_methods(fake_testing_dataset):
     """
-    must drop 11 (NAs) and then 5 more (strong signals)
-    must reset index
+    Methods must return self in order to be chained
     """
-    fake_testing_dataset.prepare_data(cols_ignore_na=[])
-    assert len(fake_testing_dataset) == 9
-    fake_testing_dataset.prepare_data(remove_strong_signals=True)
-    assert len(fake_testing_dataset) == 4
-    assert (fake_testing_dataset.data.index == list(range(4))).all()
+    fake_testing_dataset.replace_missing_data(defaults_map={"my_feature": 0}).remove_na(
```

Comment:
Here are you only testing the fact that the output of the first preprocessing function can successfully be fedto the second function of the chain? Or are you also testing that functions `replace_missing_data` and `remove_na` are functioning together?
If you're not reassigning the output of this chain to `fake_testing_dataset`, the chain isn't doing anything, right?

Reply:
It is ! both `replace_missing_data` and `remove_na` work on the data in place

Returning `self` is just a little trick in order to be able to chain methods but it does not change how the function behaves

Reply:
You _can_ catch the object returned by the method, but you don't _have_ to 

We act in_place _and_ return, which I realize now might be confusing and not a good practice...

Reply:
Ok, that makes sense!
Although it's not best practice, the only risk that I see is someone thinking a dataset is not modified (for instance during an assertion) although it actually is.
We can work on that in a next PR